nomos-services/chain-service/src/bootstrap/ibd.rs

```diff
@@ -217,13 +217,28 @@ where
                         .await;
                 }
                 DownloadsOutput::BlockReceived { block, download } => {
-                    (cryptarchia, storage_blocks_to_remove) =
-                        (self.process_block)(cryptarchia, storage_blocks_to_remove, block).await;
-                    // TODO: Stop download if process_block fails.
-                    //       (Requires refactoring the chain service)
-                    // TODO: Close the download (the underlying stream) when we need to stop it.
-                    //       https://github.com/logos-co/nomos/issues/1517
```

Comment:
Is this the current state in this PR. I think its probably missing steps. cc @youngjoon-lee